heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java

```diff
@@ -154,10 +154,40 @@ public void initialize(Config config) {
   }
 
   @Override
-  public ListenableFuture<Boolean> isTopologyRunning(String topologyName) {
+  public ListenableFuture<Boolean> doesTMasterLocationExists(
+      String topologyName) {
+    return nodeExists(getTMasterLocationPath(topologyName));
+  }
+
+  @Override
+  public ListenableFuture<Boolean> doesSchedulerLocationExists(
+      String topologyName) {
+    return nodeExists(getSchedulerLocationPath(topologyName));
+  }
+
+  @Override
+  public ListenableFuture<Boolean> doesTopologyExists(
+      String topologyName) {
     return nodeExists(getTopologyPath(topologyName));
   }
 
+  @Override
+  public ListenableFuture<Boolean> doesExecutionStateExists(
+      String topologyName) {
+    return nodeExists(getExecutionStatePath(topologyName));
+  }
+
+  @Override
+  public ListenableFuture<Boolean> doesPhysicalPlanExists(
+      String topologyName) {
+    return nodeExists(getPhysicalPlanPath(topologyName));
+  }
+
+  @Override
+  public ListenableFuture<Boolean> isTopologyRunning(String topologyName) {
```

Comment:
The semantics here are a bit loose, since we could be in a bad state where processes are running bet the state manager has state. Could we remove this method an use doesTopologyExists instead?